### DISCUSSION TITLE
基于 MMEditing 构建底层视觉算法工具体系

### DISCUSSION THREAD

**Discussion by wangruohui** [open-mmlab/mmediting#1202] [Announcements]
The issue is about the first GitLink Code Camp activity.
Related link: https://mp.weixin.qq.com/s/upVI-2TQ0ezQGpzhDiF3Pg

## 题目名称
基于 MMEditing 构建底层视觉算法工具体系

## 题目描述
MMEditing是面向底层视觉工具箱，集成了超分辨率、视频插帧、补图、抠图等方向大量SOTA模型，且持续吸纳新的底层视觉模型。一个功能统一、集中、方便用户使用的算法库将极大方便这类算法的应用。目前有如下方向可供选择：
1. 图像上色工具构建
图像上色是一个很有趣的方向，可以对黑白老照片、视频进行修复。
2. 图像补全工具构建
图像补全的目的是补全图像中缺损的地方，可以用于去水印、图像无损编辑等场景。
3. 抠图工具构建
抠图指将前景从背景中分割出来，同时可以保持边界的完整不留编辑痕迹，在视频领域有着广泛的应用。
建议选择一个方向，复现一些主流的算法，并集成到 MMEditing 工具包，同时也需要对架构设计、用户API设计提出自己的看法，最终构建一个方便开发者使用的算法工具。如果有余力也可以涉及多个方向。

## 编码任务
- 复现各个方向的主流算法（目前我们实现了一部分代码，可以以此为基础开展）
- 基于 OpenMMLab 2.0   架构对主流算法进行重构
- 将算法集成进入 MMEditing

## 技能要求和编程语言
- Python
- PyTorch
- 英文论文、技术文档的阅读能力

## 导师及联系方式
Email：[openmmlab@gmail.com](mailto:openmmlab@gmail.com)
活动微信交流群：
![](https://user-images.githubusercontent.com/15977946/169828574-a67d2692-a67d-4b8f-9297-84bd87626e59.png)

**Comment by mm-assistant** [comment]
We recommend using English or English & Chinese for issues so that we could have broader discussion.
